Formula/unisonlang.rb

```diff
@@ -0,0 +1,76 @@
+require "language/node"
+
+class Unisonlang < Formula
+  desc "Friendly programming language from the future"
+  homepage "https://unison-lang.org/"
+  url "https://github.com/unisonweb/unison.git",
+      tag:      "release/M3",
+      revision: "97cce8d23147df30b0f80ae745968db09f4e1a44"
+  version "M3"
+  license "MIT"
+  head "https://github.com/unisonweb/unison.git", branch: "trunk"
+
+  depends_on "ghc" => :build
+  depends_on "haskell-stack" => :build
+  depends_on "node" => :build
+
+  uses_from_macos "ncurses"
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "xz" => :build
+    depends_on "gmp"
+  end
+
+  resource "unison-ui" do
+    url "https://github.com/unisonweb/codebase-ui.git",
+        tag:      "release/M3",
+        revision: "05df3a235dec7d4a03ff1c588a32eb6daca288d4"
+  end
```

Comment:
Is the version here always synced with `unison`? If so, can we add a test to make sure this `resource` has the same `version` as the formula? See `ortp.rb` for an example.

Reply:
Yes, they're mutually pinned.

The test case here wouldn't be that simple unfortunately. The UI doesn't have a version number included in it anyway. The closest thing is the package.json version but it just always says `"1.0.0"`. The only thing I can think of is asserting the git tag names are the same.

Reply:
I don't see the concern as long as you add the suggested test.

Reply:
Oh, sorry, the first time I looked at the ortp formula I thought it was comparing `--version` output or something. I get it now. 😅